src/nvim/ex_docmd.c

```diff
@@ -6788,27 +6801,51 @@ void free_cd_dir(void)
 
 #endif
 
-/*
- * Deal with the side effects of changing the current directory.
- * When "local" is TRUE then this was after an ":lcd" command.
- */
-void post_chdir(int local)
+/// Deal with the side effects of changing the current directory.
+///
+/// @param scope  Scope of the function call (editor, tab or window).
+void post_chdir(int scope)
```

Comment:
@HiPhish 

1. Your logs cleary say that you have not made this function `static`.
2. Recent exploration said that you must not make it static because it is used in the API. Thus you must move `enum` definition to the header.
3. And do not use `enum cd_scope`. Use `typedef`.